/**
 * Copyright (c) 2022 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
package proto_panda;

import "assemblyDebug.proto";

message Ins {
    message IType {
        oneof type {
            int64 value_int = 1;
            double value_double = 2;
        }
    }

    uint32 opcode = 1;
    repeated uint32 regs = 2;
    repeated bytes ids = 3;
    repeated IType imms = 4;
    bytes label = 5;
    bool set_label = 6;
    DebuginfoIns ins_debug = 7;
}
